include/psen_scan/controller_msm_state_machine.h

```diff
@@ -43,111 +37,130 @@ struct stop_request_event {};
-        }
-    };
-
-    // the initial state of the player SM. Must be defined
-    typedef InitState initial_state;
-
-
-    // TODO: m is not used? Necessary?
-    typedef msm_front_ m; // makes transition table cleaner
-
-    // TODO: Use Action for function call?
-
-    // Transition table for the scanner
-    struct transition_table : mpl::vector<
-        //    Start                         Event                            Next           			 Action	                 Guard
-        //  +-------------------------------+--------------------------------+---------------------------+-----------------------+---+
-        _row < InitState,                   start_request_event,             WaitForStartReplyState                    >,
-        _row < WaitForStartReplyState,      start_reply_received_event,      WaitForMonitoringFrameState               >,
-        _row < WaitForMonitoringFrameState, monitoring_frame_received_event, WaitForMonitoringFrameState               >,
-        _row < WaitForMonitoringFrameState, stop_request_event,              WaitForStopReplyState                     >,
-        _row < WaitForStopReplyState,       stop_reply_received_event,       InitState                                 >
-        //  +-------------------------------+--------------------------------+---------------------------+-------+---+
-    > {};
-    // Replaces the default no-transition response.
-    template <class FSM,class Event>
-    void no_transition(Event const& e, FSM&,int state)
+      // std::cerr << "leaving: WaitForMonitoringFrameState\n\n";
+    }
+  };
+  struct WaitForStopReplyState : public msm::front::state<>
+  {
+    template <class Event, class FSM>
+    void on_entry(Event const&, FSM&)
     {
-        // IGNORE NON-EXISTENT TRANSITIONS FOR NOW
-        std::cout << "no transition from state " << state << " on event " << typeid(e).name() << std::endl;
+      // std::cerr << "entering: WaitForStopReplyState\n";
     }
+    template <class Event, class FSM>
+    void on_exit(Event const&, FSM&)
+    {
+      // std::cerr << "leaving: WaitForStopReplyState\n\n";
+    }
+  };
+
+  // the initial state of the player SM. Must be defined
+  typedef InitState initial_state;
+
+  typedef msm_front_ m;  // makes transition table cleaner
+
+  // Transition table for the scanner
+  struct transition_table
+    : mpl::vector<
+          //    Start                         Event                            Next           			 Action Guard
+          //  +-------------------------------+--------------------------------+---------------------------+-----------------------+---+
+          a_row<InitState,                    start_request_event,             WaitForStartReplyState,     &m::send_start_request_action>,
+          _row<WaitForStartReplyState,        start_reply_received_event,      WaitForMonitoringFrameState>,
+          _row<WaitForMonitoringFrameState,   monitoring_frame_received_event, WaitForMonitoringFrameState>,
```

Comment:
```suggestion
          _row<WaitForMonitoringFrameState,   monitoring_frame_received_event, none>,
```

However I would argue that the event `monitoring_frame_received_event` can be omitted completely.

Reply:
Y?

Reply:
See https://github.com/PilzDE/psen_scan/commit/6bf5b6f05f62cf2b2cd99eb1c253f1fee8ebc162#r41360449

Reply:
See 6bf5b6f#r41360449